### DISCUSSION TITLE
Updating PHP Environment Variable Path

### DISCUSSION THREAD

**Discussion by josealgardataborges** [leokhoa/laragon#470] [Ideas]
Is it possible that the "_Updating PHP Environment Variable Path_" menu/action, gets automatically called every time we change the PHP Version from the **LARAGON** menus?

I find it very boring, having to go manually change the PATH options in Windows Settings to reflect the new PHP version every time i change it.

Thank you.

**Comment by Pen-y-Fan** [comment]
If you use cmdr Laragon will update the PATH each time cmdr is launched.

**Reply by josealgardataborges** [reply]
Unfortunately that's not always true.

Sometimes, I have to "reload" apache and then open "cmdr" to have the correct PHP version available.

Also, if you use native "cmd" or any other command line tool, it doesn't work, you have to manually update the PATH settings.

**Reply by Pen-y-Fan** [reply]
For other terminals, you need to add Laragon to PATH:

- [How to set up Laragon on a new Windows computer (part 6) - Add to path](https://pen-y-fan.github.io/2023/01/05/how-to-set-up-laragon-on-a-new-windows-computer-part-6/)

**Comment by leokhoa** [comment]
@josealgardataborges : Have you ever tried "Add Laragon to Path"?:
![image](https://user-images.githubusercontent.com/9049575/231120894-36d03ea8-a7b3-433c-a089-a1db558f9808.png)

**Reply by Sheldon-Monk** [reply]
I've taken a few hours this morning to follow the instructions here for updating PHP and Apache. I went to PHP 8.2. VS16 x64 - Thread Safe, and Apache 2.4.57 win64 VS17. I added Laragon to the User path. I've rebooted several times.  I can now open my dev site(s) locally; however, when I rt-click on the laragon icon and open it, the PHP menu item does not have a check in front of it; and, the php 8.1.10-Win32-VS16-x64 version is the only one available.  That is the old version I was replacing with the newer 8.2.4-Win32-VS16-x64 I just downloaded & installed.  The Laragon/Menu/Apache item does have a check in front of it, and it indicates the older version of httpd-2.4.544-win64-VS16, with no listing/option of the newer version of httpd-2.4.57-win64-VS17 which I also downloaded and installed, followed by the ubiquitous Laragon start/stop operations as well as rebooting the computer.
So.  Not sure if I should call this a "win" or not.  How about you just package up a new, correct version of Laragon using the most recent versions of Apache and PHP as shown on their sites and we all call it a day.
Oh, and am I missing something?  Is there an article on medium.com that gives us the proper way to upgrade Laragon as a whole; or, should we be uninstalling an older version before installing any newer ones???
First things first though:  Even though I can now open my localhost dev site(s), I'm not sure changes will save, as I haven't made it that far yet.  Most importantly, "Why aren't the newer versions of PHP and Apache showing up and available as options in the Laragon/Menu???

**Comment by josealgardataborges** [comment]
Yes :(

For me, it stopped working correctly in V6, don't know why.

I have been using LARAGON since V4.

**Comment by Sheldon-Monk** [comment]
Based on what you've posted, I think this might be related, so thought I'd add on some info.  I too installed v.6.0.0, downloaded 4/2/2023.  Installed to a fresh Win11Pro install after Win file indexing appeared to crap out. Then again, I run 8 Tb of storage on the machine. The version of Win11Pro is:  10.0.22621, build 22621, running on AMD Ryzen 7 5800 w/64Gb RAM.
Laragon seemed to run fine for a few days then it knocked with:

PHP & Apache versions not compatible
![PHP vs Apache Versions Not Compatible-600x218x72dpi-80pct](https://user-images.githubusercontent.com/34046299/231155106-ba383e7c-ff07-464c-b016-e3318f1303a9.jpg)

I downloaded Apache 2.4.57 win32-vs16-x64, php 8.2.4 win32-v16-x64 & Visual C++ 2015-2019 14-29, as they are all the most recent stable versions.

Problem is, once I unzip all the above and try to place correctly in my current laragon install, it doesn't appear the files for the new version of Apache line-up correctly with existing laragon install:  the new Apache 2.4.57 win32-vs16-x64 has an added folder, "Apache24".  So, I'm stuck at this point and could spend far too many hours in trial and error; or, .... maybe best to revert back to an earlier laragon, but that could lead to all kinds of project loss, not to mention possible Win registry issues.

Here's the other error message I received:


![Apache Not Loading-500x343x72dpi-80pct](https://user-images.githubusercontent.com/34046299/231157798-6dea2840-2d66-49c0-b8bc-ba9e53f9c746.jpg)

And, here's a hint at the existing file structure issue in the v.6.0.0 laragon install:


![Laragon-Apache file structure-465x271x72dpi-80pct](https://user-images.githubusercontent.com/34046299/231158126-e39730b0-fb23-47c3-ab5b-5252521dbaae.jpg)

Again, not wanting to steal any of josealgardataborges thunder, or distract from his issue; just thought all this might provide further insight into the issue(s) he's having.

Any input appreciated.  Love Laragon, it solved SO MANY ISSUES we were having with failed XAMPP installs.

What's it going to take to keep Laragon alive and well???

**Reply by Pen-y-Fan** [reply]
See my blog: [How to update the PHP version in Laragon](https://pen-y-fan.github.io/2023/01/15/how-to-update-the-php-version-in-laragon/)

> What's it going to take to keep Laragon alive and well???

- [The Beauty of Laragon #450](https://github.com/leokhoa/laragon/discussions/450)
- [How to donate to Laragon? #383](https://github.com/leokhoa/laragon/discussions/383)

**Reply by josealgardataborges** [reply]
My problem, is not related to what your sharing, nevertheless, it all worked well until V6 (which I did a clean install), so no conflicts with older versions, and already formatted my windows box, so 100% sure it has nothing to do with previous v4 or v5 LARAGON installations.

@Sheldon-Monk The problems you were facing are related to the compiler, which has to match between PHP and Apache (VS19, VS17, VS16, etc).

**Reply by Sheldon-Monk** [reply]
So, how do I fix that?  Have been an MS-guy my entire life, so need some complete instructions.  What about that extra Apache24 folder.  How do I deal with that?

**Reply by josealgardataborges** [reply]
I have lots of apache versions installed, and as long as I select the "same" "Compiler" (32/64bits) between apache and php, it all works well.

![image](https://user-images.githubusercontent.com/8929655/231191720-32318eef-82b2-44b2-bdf2-86be7c4e2e61.png)

![image](https://user-images.githubusercontent.com/8929655/231191788-9cfbbee6-8e52-447c-8335-74c90f59de6f.png)

You just need to "match" PHP VC11 to VC11 apache, PHP VS16 to VS16 apache, etc

**Reply by josealgardataborges** [reply]
For instance I can only use PHP "php-5.4.14-Win32-VC9-x86" with APACHE "httpd-2.4.6-win32-VC9"

**Reply by Sheldon-Monk** [reply]
I posted this above and now realize I should have put it down here.  Apologies!  Here you go:
I've taken a few hours this morning to follow the instructions here for updating PHP and Apache. I went to PHP 8.2. VS16 x64 - Thread Safe, and Apache 2.4.57 win64 VS17. I added Laragon to the User path. I've rebooted several times. I can now open my dev site(s) locally; however, when I rt-click on the laragon icon and open it, the PHP menu item does not have a check in front of it; and, the php 8.1.10-Win32-VS16-x64 version is the only one available. That is the old version I was replacing with the newer 8.2.4-Win32-VS16-x64 I just downloaded & installed. The Laragon/Menu/Apache item does have a check in front of it, and it indicates the older version of httpd-2.4.544-win64-VS16, with no listing/option of the newer version of httpd-2.4.57-win64-VS17 which I also downloaded and installed, followed by the ubiquitous Laragon start/stop operations as well as rebooting the computer.
So. Not sure if I should call this a "win" or not. How about you just package up a new, correct version of Laragon using the most recent versions of Apache and PHP as shown on their sites and we all call it a day.
Oh, and am I missing something? Is there an article on medium.com that gives us the proper way to upgrade Laragon as a whole; or, should we be uninstalling an older version before installing any newer ones???
First things first though: Even though I can now open my localhost dev site(s), I'm not sure changes will save, as I haven't made it that far yet. Most importantly, "Why aren't the newer versions of PHP and Apache showing up and available as options in the Laragon/Menu???

**Reply by Sheldon-Monk** [reply]
@test-algardata, I'll give that a go.  Unfortuantely, PHP and Apache in there curretnly posted versions are using different VS'.  I'll backup on one or the other and see if that remedies it all.  (As it stands right now, neither app is showing the most recent version to be able to switch to in the Laragon/Menu.  Later.

**Reply by Pen-y-Fan** [reply]
> So, how do I fix that? Have been an MS-guy my entire life, so need some complete instructions. What about that extra Apache24 folder. How do I deal with that?

Use windows explorer to navigate into the extra Apache24 folder, select all (CTRL + A) and cut (CTRL + X), navigate to the previous folder and paste (CTRL + C). Laragon will automatically detect the Apache files. You may need to stop all and start.

The extra Apache24 folder can be deleted.

When moved the folder structure should look like this:

![image](https://user-images.githubusercontent.com/40126936/231242655-7c4dd139-fb20-444c-8b74-3533e1320cf7.png)

**Reply by josealgardataborges** [reply]
Yes this is the correct folder structure (both for PHP or apache):
![Apache](https://user-images.githubusercontent.com/8929655/231418872-7e2c1228-9ff8-4844-8dd9-e00b4352bd03.png)
![PHP](https://user-images.githubusercontent.com/8929655/231418895-ec075a24-b7e1-4235-83c9-24e7d86a4267.png)

The name of the "main" folder is "irrelevant" (it's what LARAGON will show on the menu) for it to work well, what needs to be correct is the "inside" of that folder structure, as seen on the above pictures.

**Reply by josealgardataborges** [reply]
You can have as many versions of PHP, APACHE or MYSQL as long as they are in the correct place LARAGON will catch them and allow you to use them.

Rule of thumb, on "root" directory for each, and them all the relevant files inside it:

**GOOD:**
C:\Laragon\bin\php\php-5.4.14-Win32-VC9-x86
![image](https://user-images.githubusercontent.com/8929655/231420569-261c499b-60a7-4406-9ee8-364c0eedab29.png)


**BAD**
C:\Laragon\bin\php\php-5.4.14-Win32-VC9-x86\php-5.4.14-Win32-VC9-x86
![image](https://user-images.githubusercontent.com/8929655/231420784-b78e2583-a075-44d0-ac81-bda89fd4650a.png)

**Reply by josealgardataborges** [reply]
Follow closely this tutorial: https://medium.com/@leovn/how-to-add-another-php-version-to-laragon-a0f28ee32fd4

and this: https://medium.com/@leovn/how-to-add-another-apache-version-to-laragon-5a25d4516938

**Comment by Sheldon-Monk** [comment]
(I don't mind donating, but not seeing a link on the laragon.org site to do so.)  Think how nice it would be if ALL OF US donated $20......

You have to admit, we get far more than that in VALUE from it.

**Reply by Pen-y-Fan** [reply]
[How to donate to Laragon? #383](https://github.com/leokhoa/laragon/discussions/383)

**Comment by sirhvd** [comment]
Changing the PHP, node version does not update the environment variable path.
I'm using Laragon 6.0.0 and have to re-run the "Add Laragon to Path" every time I change the PHP, node version.
